contracts/sol6/Dao/mock/KyberTokenFeeHandler.sol

```diff
@@ -0,0 +1,582 @@
+                    sanity rate.
+ *              Platfrom fee: accumulated per platform wallet, can be claimed any time.
+ *      2. Network Fee distribution: Per epoch kyberFeeHandler contract reads BRR distribution percentage 
+ *          from kyberDao. When the data expires, kyberFeeHandler reads updated values.
+ */
+contract KyberTokenFeeHandler is IKyberFeeHandler, Utils5, DaoOperator, ReentrancyGuard {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    uint256 internal constant DEFAULT_REWARD_BPS = 3000;
+    uint256 internal constant DEFAULT_REBATE_BPS = 3000;
+    uint256 internal constant SANITY_RATE_DIFF_BPS = 1000; // 10%
+
+    struct BRRData {
+        uint64 expiryTimestamp;
+        uint32 epoch;
+        uint16 rewardBps;
+        uint16 rebateBps;
+    }
+
+    struct BRRWei {
+        uint256 rewardTwei;
+        uint256 rebateTwei;
+        uint256 burnTwei;
+    }
+
+    IKyberDao public kyberDao;
+    IKyberProxy public kyberProxy;
+    address public kyberNetwork;
+    IERC20 public immutable quoteToken;
+    IERC20 public immutable knc;
+
+    uint256 public immutable burnBlockInterval;
+    uint256 public lastBurnBlock;
+
+    BRRData public brrAndEpochData;
+    address public daoSetter;
+
+    /// @dev amount of Twei to burn for each burn knc call
+    uint256 public tweiToBurn;
+
+    mapping(address => uint256) public feePerPlatformWallet;
+    mapping(address => uint256) public rebatePerWallet;
+    mapping(uint256 => uint256) public rewardsPerEpoch;
+    mapping(uint256 => uint256) public rewardsPaidPerEpoch;
+    // hasClaimedReward[staker][epoch]: true/false if the staker has/hasn't claimed the reward for an epoch
+    mapping(address => mapping (uint256 => bool)) public hasClaimedReward;
+    uint256 public totalPayoutBalance; // total balance in the contract that is for rebate, reward, platform fee
+
+    /// @dev use to get rate of KNC/ETH to check if rate to burn knc is normal
+    /// @dev index 0 is currently used contract address, indexes > 0 are older versions
+    ISanityRate[] internal sanityRateContract;
+
+    event FeeDistributed(
+        IERC20 indexed token,
+        address indexed platformWallet,
+        uint256 platformFeeTwei,
+        uint256 rewardTwei,
+        uint256 rebateTwei,
+        address[] rebateWallets,
+        uint256[] rebatePercentBpsPerWallet,
+        uint256 burnAmtTwei
+    );
+
+    event BRRUpdated(
+        uint256 rewardBps,
+        uint256 rebateBps,
+        uint256 burnBps,
+        uint256 expiryTimestamp,
+        uint256 indexed epoch
+    );
+
+    event EthReceived(uint256 amount);
+    event KyberDaoAddressSet(IKyberDao kyberDao);
+    event BurnConfigSet(ISanityRate sanityRate, uint256 tweiToBurn);
+    event RewardsRemovedToBurn(uint256 indexed epoch, uint256 rewardsTwei);
+    event KyberNetworkUpdated(address kyberNetwork);
+    event KyberProxyUpdated(IKyberProxy kyberProxy);
+
+    constructor(
```

Comment:
Since KyberDao doesn't need FeeHandler anymore, we can remove daoSetter and add daoAddress directly